src/dags/gob.py

```diff
@@ -42,63 +59,94 @@ def create_gob_dag(is_first, gob_dataset_name, gob_table_name):
                 extra_kwargs["endpoint"] = GOB_SECURE_ENDPOINT
 
     dag = DAG(
-        f"{dag_id}_{gob_db_table_name}",
+        f"{dag_id}_{dataset_table_id}",
         default_args={"owner": owner, **default_args},
         schedule_interval=f"0 {schedule_start_hour} * * *" if is_first else None,
         tags=["gob"],
     )
 
     kwargs = dict(
-        task_id=f"load_{gob_db_table_name}",
+        task_id=f"load_{dataset_table_id}",
         endpoint=GOB_PUBLIC_ENDPOINT,
-        dataset=gob_dataset_name,
-        table_name=gob_table_name,
         retries=3,
         graphql_query_path=graphql_dir_path / "query.graphql",
         max_records=MAX_RECORDS,
         http_conn_id="gob_graphql",
         token_expires_margin=OAUTH_TOKEN_EXPIRES_MARGIN,
+        xcom_table_info_task_ids=f"mkinfo_{dataset_table_id}",
     )
 
     with dag:
 
         # 1. Post info message on slack
         slack_at_start = MessageOperator(
-            task_id=f"slack_at_start_{gob_db_table_name}",
+            task_id=f"slack_at_start_{dataset_table_id}",
             http_conn_id="slack",
             webhook_token=slack_webhook_token,
             message=f"Starting {dag_id} ({DATAPUNT_ENVIRONMENT})",
             username="admin",
         )
 
-        # 2. drop temp table if exists
+        def _create_dataset_info(dataset_id: str, table_id: str) -> DatasetInfo:
+            dataset = schema_def_from_url(SCHEMA_URL, dataset_id)
+            # Fetch the db_name for this dataset and table
+            db_table_name = dataset.get_table_by_id(table_id).db_name()
+            # provide full qualified name, for convenience
+            dataset_table_id = f"{dataset_id}_{table_id}"
+            return DatasetInfo(dataset, table_id, dataset_table_id, db_table_name)
+
+        # 2. Create Dataset info to put on the xcom channel for later use
+        # by operators
+        create_dataset_info = PythonOperator(
+            task_id=f"mkinfo_{dataset_table_id}",
+            python_callable=_create_dataset_info,
+            op_args=(gob_dataset_id, gob_table_id),
+        )
+
+        def init_assigner(o: Any, x: Any) -> None:
```

Comment:
Maybe this can be a little more specific. Setting the attribute always seems to happen on a `PostgresOperator` derived class. Hence you'd probably want at least:
```python
def init_assigner(o: PostgresOperator, x: Any) -> None:
   ...
```
However MyPy will not like that, as dynamically setting an attribute on a class that doesn't define that attribute is technically not type correct. That is, unless you define `__setattr__` specifically. But that means defining an intermediate base class:

```python
class AttrPostgresOperator(PostgresOperator):
    """"Allow any property to be get and set in MyPy compatible way."""
    def __setattr__(self, name: str, value: Any) -> None:
        super().__setattr__(name, value)

    def __getattr__(self, name: str) -> Any:
        return super().__getattribute__(name)

class PostgresTableInitOperator(AttrPostgresOperator):
    ...
```
Which in turn means that the type annotation on `init_assigner` becomes:
```python
def init_assigner(o: AttrPostgresOperator, x: Any) -> None:
   ...
```
That might be a bit too much. Anyway got this from https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html#when-you-re-puzzled-or-when-things-are-complicated (see `class A` and the comments)